Plugins/cmake-smoke-test/cmake-smoke-test.swift

```diff
@@ -182,7 +186,53 @@ extension Process {
     }
 
     static func checkNonZeroExit(url: URL, arguments: [String], workingDirectory: URL, environment: [String: String]? = nil) async throws {
-        print("\(url.path()) \(arguments.joined(separator: " "))")
+        Diagnostics.progress("\(url.path()) \(arguments.joined(separator: " "))")
+        #if USE_PROCESS_SPAWNING_WORKAROUND
+        Diagnostics.progress("Using process spawning workaround")
+        // Linux workaround for https://github.com/swiftlang/swift-corelibs-foundation/issues/4772
+        // Foundation.Process on Linux seems to inherit the Process.run()-calling thread's signal mask, creating processes that even have SIGTERM blocked
+        // This manifests as CMake hanging when invoking 'uname' with incorrectly configured signal handlers.
+        var fileActions = posix_spawn_file_actions_t()
+        defer { posix_spawn_file_actions_destroy(&fileActions) }
+        var attrs: posix_spawnattr_t = posix_spawnattr_t()
+        defer { posix_spawnattr_destroy(&attrs) }
+        posix_spawn_file_actions_init(&fileActions)
+        posix_spawn_file_actions_addchdir_np(&fileActions, workingDirectory.path())
+
+        posix_spawnattr_init(&attrs)
+        posix_spawnattr_setpgroup(&attrs, 0)
+        var noSignals = sigset_t()
+        sigemptyset(&noSignals)
+        posix_spawnattr_setsigmask(&attrs, &noSignals)
+
+        var mostSignals = sigset_t()
+        sigemptyset(&mostSignals)
+        for i in 1 ..< SIGSYS {
```

Comment:
`..<SIGRTMAX`?

Reply:
I don't think anything is using the realtime signals, and there could be a gap between SIGSYS and SIGRTMIN on some implementations